### ISSUE TITLE
Gulp.src glob pattern fails in various edge situations

### ISSUE THREAD

**Issue by proton-ab** [gulpjs/gulp#2783]
Before you open this issue, please complete the following tasks:

* [x] use the search bar at the top of the page to search this repository for similar issues or discussions that have already been opened.
* [x] if you are looking for help from the gulp team or community, open a [discussion](https://github.com/gulpjs/gulp/discussions).
* [x] if you think there is a problem with the plugin you're using, open a [discussion](https://github.com/gulpjs/gulp/discussions).
* [x] if you think there is a bug in our code, open this issue.

### What were you expecting to happen?

`Gulp.src` works as it did previously in v4, ignoring directories and files outside of scope of patterns provided. No error is raised.

### What actually happened?

`Gulp.src` scans entire directory tree and fails on at least these cases:
- Another process runs in different sub-directory that creates and removes temporary files
- Broken or invalid link is encountered

### Please give us a sample of your gulpfile
```js
import path from 'path';
import os from 'os';

// Gulp specific
import Gulp from 'gulp';

const dirs = {
  tmp: path.join(os.tmpdir(), 'gulp-build-cache'),
  out: 'public/static',
};

// ---

function hash() {
  return Gulp.src(`${dirs.tmp}/**/*.*`)
    .pipe(Gulp.dest(dirs.out))
}

// ---

function css() {
  return Gulp.src('assets/public/**/*.css')
    .pipe(Gulp.dest(dirs.tmp));
}

// ---

export const build = Gulp.series(css, hash);
```

### Terminal output / screenshots

```sh
vagrant@debian11:/code$ yarn gulp build
yarn run v1.22.22
$ node ./node_modules/gulp/bin/gulp.js build
[16:26:34] Using gulpfile /code/gulpfile.mjs
[16:26:34] Starting 'build'...
[16:26:34] Starting 'css'...
[16:26:40] Finished 'css' after 6.07 s
[16:26:40] Starting 'hash'...
[16:26:42] 'hash' errored after 2.12 s
[16:26:42] Error: ENOENT: no such file or directory, scandir '/vendor/dflydev/fig-cookies'
[16:26:42] 'build' errored after 8.2 s
```

```
vagrant@debian11:/code$ yarn gulp build
yarn run v1.22.22
$ node ./node_modules/gulp/bin/gulp.js build
[16:31:32] Using gulpfile /code/gulpfile.mjs
[16:31:32] Starting 'build'...
[16:31:32] Starting 'css'...
[16:31:32] 'css' errored after 20 ms
[16:31:32] Error: ENOENT: no such file or directory, stat '/code/scripts'
[16:31:32] 'build' errored after 23 ms
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
vagrant@debian11:/code$ ls -l /code/scripts
lrwxrwxr-x 1 vagrant www-data 0 Oct 29  2021 /code/scripts -> ../calamari/scripts
```

### Please provide the following information:
* OS & version [e.g. MacOS Catalina 10.15.4]: Debian GNU/Linux 11
* node version (run `node -v`): v18.20.1
* npm version (run `npm -v`): 10.5.0
* gulp version (run `gulp -v`): 5.0.0 (CLI 3.0.0)

### Additional information
The first terminal output is result of running `composer` in parallel with `gulp`. Second one is result of placing broken link in directory tree (the link becomes valid later on, but is not at the time `gulp` is being run). The issue seems related to https://github.com/gulpjs/gulp/issues/2771#issuecomment-2030985951 but is different enough that I believe it warrants raising separate issue.
